doc/book/helpers/server-url-helper.md

```diff
@@ -78,20 +79,22 @@ $container->set(
 );
 ```
 
-To register the `ServerUrlMiddleware` as pre-routing pipeline middleware:
+To register the `ServerUrlMiddleware` as pipeline middleware following the
+routing middleware:
 
 ```php
 use Zend\Expressive\Helper\ServerUrlMiddleware;
 
-// Do this early, before piping other middleware or routes:
+// Programmatically:
 $app->pipe(ServerUrlMiddleware::class);
+$app->pipeRoutingMiddleware();
+$app->pipeDispatchMiddleware();
 
 // Or use configuration:
 // [
 //     'middleware_pipeline' => [
-//         'pre_routing' => [
-//             ['middleware' => ServerUrlMiddleware::class],
-//         ],
+//         ['middleware' => ServerUrlMiddleware::class, 'priority' => PHP_INT_MAX],
```

Comment:
@danizord It may or may not be; I need to look up how PHP stores integers (i.e., if it allocates enough bytes to allow for the maximum, or only the amount needed for the integer itself). However, you're wrong about this:

> Also, as said above it should be piped before routing, so PHP_INT_MAX won't work.

Order is defined by using a *priority queue*; as such, *higher* numbers indicate that the middleware is piped to the application *earlier*. So this is correct here.